sympy/core/intfunc.py

```diff
@@ -0,0 +1,559 @@
+import math
+import sys
+from functools import lru_cache
+
+from .sympify import sympify
+from .singleton import S
+from sympy.external.gmpy import gcd as number_gcd, lcm as number_lcm, sqrt, iroot
+from sympy.utilities.misc import as_int, filldedent
+
+from mpmath.libmp import bitcount
+
+
+def ndigits(n, base=10):
+    """Return the number of digits needed to express n in give base.
+
+    Examples
+    ========
+
+    >>> from sympy.ntheory.digits import ndigits
+    >>> ndigits(10)
+    2
+    >>> ndigits(10, 2)  # 1010 -> 4 digits
+    4
+    >>> ndigits(-100, 16)  # -64 -> 2 digits
+    2
+
+
+    Parameters
+    ==========
+
+    n: integer
+        The number whose digits are counted.
+
+    b: integer
+        The base in which digits are computed.
+
+    See Also
+    ========
+    ntheory.digits.digits, ntheory.digits.count_digits
+    """
+    if base < 0:
+        raise ValueError('base must be int greater than 1')
+    if not n:
+        return 1
+    e, t = integer_log(abs(n), base)
+    return 1 + e
+
+
+def integer_log(n, b):
+    r"""
+    Returns ``(e, bool)`` where e is the largest nonnegative integer
+    such that :math:`|n| \geq |b^e|` and ``bool`` is True if $n = b^e$.
+
+    Examples
+    ========
+
+    >>> from sympy import integer_log
+    >>> integer_log(125, 5)
+    (3, True)
+    >>> integer_log(17, 9)
+    (1, False)
+    >>> integer_log(4, -2)
+    (2, True)
+    >>> integer_log(-125,-5)
+    (3, True)
+
+    See Also
+    ========
+    integer_nthroot
+    sympy.ntheory.primetest.is_square
+    sympy.ntheory.factor_.multiplicity
+    sympy.ntheory.factor_.perfect_power
+    """
+    if b < 0:
+        e, t = integer_log(n if n > 0 else -n, -b)
+        if n < 0:
+            t = t and e % 2 == 1
+        else:
+            t = t and bool(n % 2 if n < 0 else not n % 2)
```

Comment:
```suggestion
            t = t and e % 2 == 0
```
Perhaps a mistake in the modification.